### DISCUSSION TITLE
Can't find server file when start Remix + Vite application

### DISCUSSION THREAD

**Discussion by pauloricardos** [remix-run/remix#9128] [Q&A]
I'm having the following issue when I start my app: `Error: ENOENT: no such file or directory, stat '/../build/server/index.js'`

This is the build output:
 ![Screenshot 2024-03-25 at 10 10 26](https://github.com/remix-run/remix/assets/40395411/2d83b61f-2e0b-4fa5-b3a0-712191171061)
 
 I noticed the server index file is being generated in a different path from the expected. Should be `server/index.js` instead?
 
 I'm using Bun version 1.0.35
 
 This is my repo: https://github.com/pauloricardos/pokedex-web-app
 
 Already removed the `bun.lockb`, `build` folder and `node_modules` and tried to build and start again.

**Comment by joaoguidev** [comment]
Hello.

I'm not sure how it works on Bun but maybe that might help you. [See here](https://remix.run/docs/en/main/future/vite#new-build-output-paths)

I was having problem deploying my Vite app to Cloudflare and solved by changing the output directory to `./build/client` instead of `./public`
[Discussion here](https://github.com/remix-run/remix/discussions/9099)

**Reply by pauloricardos** [reply]
thanks for helping, but i think it's a different problem

**Comment by RabeeAbuBaker** [comment]
I'm facing the same problem here (not using Bun though), any luck fixing this?

<img width="1252" alt="image" src="https://github.com/remix-run/remix/assets/38038348/f2a5e770-91f8-4039-abf2-f7c67e357b60">

**Reply by pauloricardos** [reply]
@RabeeAbuBaker i gave up on running my application on production mode. i have no clue how this node-js folder is built. but i could deploy my app successfully by outputing the `build/server` folder. may changing `remix-serve ./build/server/index.js` to `remix-serve ./build/server/<node-js-folder-here>/index.js` can work

**Comment by styxlab** [comment]
Running into the same issue. In my case this is caused by the vercel presets. My solution is to use the vercel presets only in the vercel environment:
```
import { vitePlugin as remix } from "@remix-run/dev";
import { defineConfig } from "vite";
import tsconfigPaths from "vite-tsconfig-paths";
import { vercelPreset } from '@vercel/remix/vite';

const isVercel = process.env.VERCEL === "1";

export default defineConfig({
  plugins: [
    remix({
      ...isVercel && { presets: [vercelPreset()] },
      future: {
        v3_fetcherPersist: true,
        v3_relativeSplatPath: true,
        v3_throwAbortReason: true,
      },
    }),
    tsconfigPaths(),
  ],
});

```

**Reply by lucastnr** [reply]
Thank you, helpful until this day
